### ISSUE TITLE
getCgroupV2CpuStats get exception when parse unsigned 64 bit number

### ISSUE THREAD

**Issue by yummypeng** [elastic/elasticsearch#112274]
### Elasticsearch Version

v8.15.0

### Java Version

openjdk version "22.0.1" 2024-04-16

### OS Version

Linux with kernel 5.10 (CGroupv2)

### Problem Description

The Linux kernel stores CPU stats as 64-bit unsigned integers, but `getCgroupV2CpuStats()` reads them as long, possibly causing unexpected exceptions.
https://github.com/torvalds/linux/blob/master/kernel/cgroup/rstat.c#L631

![Image](https://github.com/user-attachments/assets/3f8602f6-1afb-426e-a6bf-5529b487393e)

### Steps to Reproduce

1.
2.
3.

### Logs (if relevant)

```
java.lang.NumberFormatException: For input string: \"18446744073709551615\"
	at java.base/java.lang.NumberFormatException.forInputString(NumberFormatException.java:67)
	at java.base/java.lang.Long.parseLong(Long.java:618)
	at java.base/java.lang.Long.parseLong(Long.java:722)
	at org.elasticsearch.server@8.14.2/org.elasticsearch.monitor.os.OsProbe.getCgroupV2CpuStats(OsProbe.java:639)
	at org.elasticsearch.server@8.14.2/org.elasticsearch.monitor.os.OsProbe.getCgroup(OsProbe.java:693)
	at org.elasticsearch.server@8.14.2/org.elasticsearch.monitor.os.OsProbe.getCgroup(OsProbe.java:889)
	at org.elasticsearch.server@8.14.2/org.elasticsearch.monitor.os.OsProbe.osStats(OsProbe.java:896)
	at org.elasticsearch.ml@8.14.2/org.elasticsearch.xpack.ml.MachineLearning.additionalSettings(MachineLearning.java:844)
	at org.elasticsearch.server@8.14.2/org.elasticsearch.node.Node.mergePluginSettings(Node.java:776)
	at org.elasticsearch.server@8.14.2/org.elasticsearch.node.NodeConstruction.createEnvironment(NodeConstruction.java:453)
	at org.elasticsearch.server@8.14.2/org.elasticsearch.node.NodeConstruction.prepareConstruction(NodeConstruction.java:252)
	at org.elasticsearch.server@8.14.2/org.elasticsearch.node.Node.<init>(Node.java:192)
	at org.elasticsearch.server@8.14.2/org.elasticsearch.bootstrap.Elasticsearch$2.<init>(Elasticsearch.java:240)
	at org.elasticsearch.server@8.14.2/org.elasticsearch.bootstrap.Elasticsearch.initPhase3(Elasticsearch.java:240)
	at org.elasticsearch.server@8.14.2/org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:75)
```

**Comment by yummypeng** [comment]
@rjernst I actually hit this bug due to adding a new u64 CPU statistic in cpu.stat, aligning with existing stats. Would Elasticsearch consider adapting to resolve this issue, given that some metrics like MemoryLimitInBytes are parsed as BigInteger?

**Comment by SouradipPoddar** [comment]
@rjernst I have worked on this issue and submitted a pull request. Please take a look.

**Comment by kunal5042** [comment]
@rjernst the pull request submitted by @SouradipPoddar has been merged, I think the issue has been resolved and can be closed, kindly verify the same and update the status. 
Thanks.

**Comment by ldematte** [comment]
Yes, this have been fixed for 8.17 https://github.com/elastic/elasticsearch/pull/115915 and 9.0 https://github.com/elastic/elasticsearch/pull/114681. Closing.
